File: app/src/main/java/net/cynreub/subly/ui/subscriptions/detail/SubscriptionDetailUiState.kt
package net.cynreub.subly.ui.subscriptions.detail

import net.cynreub.subly.domain.model.PaymentMethod
import net.cynreub.subly.domain.model.Subscription

data class SubscriptionDetailUiState(
    // Core Data
    val subscription: Subscription? = null,
    val paymentMethod: PaymentMethod? = null,

    // UI State
    val isLoading: Boolean = true,
    val error: String? = null,

    // Action States
    val isDeleting: Boolean = false,
    val isMarkingAsPaid: Boolean = false,
    val showDeleteDialog: Boolean = false,

    // Success State (for navigation)
    val deleteSuccess: Boolean = false
)

File: subly_ios/Subly/Subly/UI/Subscriptions/Detail/SubscriptionDetailUiState.swift
import Foundation

struct SubscriptionDetailUiState {
    var subscription: Subscription? = nil
    var category: Category? = nil
    var paymentMethod: PaymentMethod? = nil

    var isLoading: Bool = true
    var error: String? = nil

    var isDeleting: Bool = false
    var isTogglingActive: Bool = false
    var showDeleteConfirmation: Bool = false

    var deleteSuccess: Bool = false
}
